src/pages/[id].js

```diff
@@ -34,6 +35,13 @@ export default function CallDetail() {
   const { id } = router.query;
   const [call, setCall] = useState([]);
   const [applicants, setApplicants] = useState();
+  const [applicationStatus, setApplicationStatus] = useState();
+
+  useEffect(() => {
+    setApplicationStatus(
+      profileData?.appliedCalls?.find((call) => call.id === id)?.status
+    );
+  }, []);
```

Comment:
Let's add `profileData` in the dependency array

I will bring eslint react hooks rules asap, this will help us